### PR TITLE
Xarray change, allowing for multiple `*` selectors in the variable path

### PR DESCRIPTION
**Pull request by v1kko**
- Created an example with ions in a notebook
- Removed `$<var>` dimension selector and replaced with `*`
- Updated tests and notebooks

### CODE REVIEW

**Comment by v1kko** [comment]
Hi Stef, in the end I stepped away from the `coordinate` idea, but I did improve the other things we discussed. Let me know what you think

File: src/duqtools/ids/_mapping.py [hunk 1]
```diff
@@ -356,6 +354,18 @@ def to_dataframe(self,
...
+        for index in range(len(self[partial_path[0]])):
+            path = partial_path[0] + '/' + str(index) + '/' + partial_path[1]
+            if len(partial_path) > 2:
+                sub_partial_path = [path] + partial_path[2:]
+                sub_arr = self._fill_array_from_partial_path(sub_partial_path)
+                arr.append(sub_arr)
+            else:
+                arr.append(self[path])
+        return arr
+
```
Comment (stefsmeets) [inline src/duqtools/ids/_mapping.py:368]: I found this function a bit tricky to read, so I had some fun trying to refactor while trying to understand how it works. Hope you don't mind :-)

```suggestion
    def _fill_array_from_parts(self, *parts: str):
        arr = []
        root, sub, *remaining = parts
        nodes = self[root]
        
        for index in range(len(nodes)):
            path = f'{root}/{index}/{sub}'
            
            if remaining:
                sub_arr = self._fill_array_from_parts(path, *remaining)
            else:
                sub_arr = self[path]
            
            arr.append(sub_arr)

        return arr
```

File: src/duqtools/ids/_mapping.py [hunk 2]
```diff
@@ -378,27 +388,17 @@ def to_xarray(
         xr_data_vars: Dict[str, Tuple[List[str], np.array]] = {}
 
         for var in variables:
-            dimensions = DIM_PATTERN.findall(var.path)
+            partial_path = var.path.split('/*/')
```
Comment (stefsmeets) [inline src/duqtools/ids/_mapping.py:391]:
```suggestion
            parts = var.path.split('/*/')
```

File: src/duqtools/ids/_mapping.py [hunk 3]
```diff
@@ -378,27 +388,17 @@ def to_xarray(
...
 
-            if not dimensions:
+            if len(partial_path) == 1:
```
Comment (stefsmeets) [inline src/duqtools/ids/_mapping.py:393]:
```suggestion
            if len(parts) == 1:
```

File: src/duqtools/ids/_mapping.py [hunk 4]
```diff
@@ -378,27 +388,17 @@ def to_xarray(
...
-
-            index_string, dimension = dimensions[0]
-            prefix = var.path.split(index_string)[0].strip('/')
-
-            arr = []
-
-            for index in range(len(self[prefix])):
-                path = var.path.replace(index_string, str(index))
-                arr.append(self[path])
+            arr = self._fill_array_from_partial_path(partial_path)
```
Comment (stefsmeets) [inline src/duqtools/ids/_mapping.py:397]:
```suggestion
            arr = self._fill_array_from_partial_path(*parts)
```

**Review by stefsmeets — APPROVED** [review]
Cool, I love how simple this `to_xarray()` method is now 👍 This means we can probably get rid of some of the regex functions soon and `to_numpy()` etc.

Cheers for the notebook too!

I also realized yesterday that the `$var` setup would be a perfect use case for [templates](https://docs.python.org/3/library/string.html#template-strings), but I guess it is too late now 😅

File: src/duqtools/ids/_mapping.py (hunk 1 above)
Comment (v1kko) [reply src/duqtools/ids/_mapping.py:368]: So much more pythonic, It looks much better,  I guess I am really more used in doing thing the "old" way.

**Pull request merged**